src/module.c

```diff
@@ -152,25 +173,31 @@ int MGraph_Query(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     AST_Query* ast = _parse_query(ctx, query, graphName, &errMsg);
     if (!ast) return REDISMODULE_OK;
 
-    ExecutionPlan *plan = NewExecutionPlan(ctx, graphName, ast);
-    ResultSet* resultSet = ExecutionPlan_Execute(plan);
-    /* Send result-set back to client. */
-    ExecutionPlanFree(plan);
-    ResultSet_Replay(ctx, resultSet);
-
-    /* Replicate query only if it modified the keyspace. */
-    if(Query_Modifies_KeySpace(ast) &&
-            (resultSet->labels_added > 0 ||
-             resultSet->nodes_created > 0 ||
-             resultSet->properties_set > 0 ||
-             resultSet->relationships_created > 0 ||
-             resultSet->nodes_deleted > 0 ||
-             resultSet->relationships_deleted > 0)) {
-        RedisModule_ReplicateVerbatim(ctx);
+    if (ast->indexNode != NULL) { // index operation
+        int success = _index_operation(ctx, graphName, ast->indexNode);
+        // return from this function if we have already enqueued a reply
+        if (!success) return REDISMODULE_OK;
```

Comment:
No need to check return value (as an error was already emitted),
Simply return REDISMODULE_OK  here.

Reply:
This logic is being used because we will have only emitted an error if the operation failed. If it was successful, we will exit this conditional and send the same RedisModule_Reply from this function as an execution plan would: 

```
127.0.0.1:7777> GRAPH.QUERY imdb "DROP INDEX ON :actor(age)"
(error) Redis-Graph only supports index creation operations at present.

127.0.0.1:7777> GRAPH.QUERY imdb "CREATE INDEX ON :actor(age)"
"Query internal execution time: 2.099000 milliseconds"
```

Reply:
Alright, need to see what type of response Neo is returning for index operations.